### PR TITLE
New contributions project: TrackingRegion

### PR DESCRIPTION
**Pull request by zhangevan**
This project contains 4 client components that you can use in your clients.

Client Components:
com.vantiq.component.ui.BoundaryEditor - A eitor for specifying boundary (an array of 4 points) over an selected image. com.vantiq.component.ui.DirectionDistanceEditor - A editor for editing "distance" or "direction" property of a TrackingRegion. com.vantiq.component.ui.ConvertCoordinates - A editor for editing "convertionSepcification" property of a TrackingRegion. com.vantiq.component.ui.TrackingRegionEditor - A Client component to create/view/edit a tracking region. It use above three client components to edit its properties. 

Using the Tracking Region Editor
You can run com.vantiq.component.ui.TrackingRegionEditor as a normal Client to create/edit/view tracking regions. You can also embed the editor as a component to your Client. See TestTrackingRegion as an example. The TrackingRegionEditor component's toolbar buttons are hidden by setting its hideToolbar property to true.

### CODE REVIEW

File: TrackingRegion/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+### Client Components: ###
+* com.vantiq.component.ui.BoundaryEditor - A eitor for specifying boundary (an array of 4 points) over an selected image.
+* com.vantiq.component.ui.DirectionDistanceEditor - A editor for editing "distance" or "direction" property of a TrackingRegion.
+* com.vantiq.component.ui.ConvertCoordinates - A editor for editing "convertionSepcification" property of a TrackingRegion.
+* com.vantiq.component.ui.TrackingRegionEditor - A Client component to create/view/edit a tracking region. It use above three client components to edit its properties.  
+
+### Using the Tracking Region Editor ###
+You can run com.vantiq.component.ui.TrackingRegionEditor as a normal Client to create/edit/view tracking regions.
+You can also embed the editor as a component to your Client. See TestTrackingRegion as an example. The TrackingRegionEditor component's toolbar buttons are hidden by setting its hideToolbar property to true.
```
Comment (rblumer4) [inline TrackingRegion/README.md]: add some italics to highlight settings
```suggestion
You can also embed the editor as a component to your Client. See _TestTrackingRegion_ as an example. The TrackingRegionEditor component's toolbar buttons are hidden by setting its _hideToolbar_ property to `true`.
```

File: TrackingRegion/README.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+This project  contains 4 client components that you can use in your clients.
+
+### Client Components: ###
+* com.vantiq.component.ui.BoundaryEditor - A eitor for specifying boundary (an array of 4 points) over an selected image.
+* com.vantiq.component.ui.DirectionDistanceEditor - A editor for editing "distance" or "direction" property of a TrackingRegion.
+* com.vantiq.component.ui.ConvertCoordinates - A editor for editing "convertionSepcification" property of a TrackingRegion.
+* com.vantiq.component.ui.TrackingRegionEditor - A Client component to create/view/edit a tracking region. It use above three client components to edit its properties.
```
Comment (rblumer4) [inline TrackingRegion/README.md]: typo: eitor -> editor  
typo A -> An   (on all 4 lines)
Also, make component names bold or italic, e.g.
```suggestion
* **com.vantiq.component.ui.BoundaryEditor** - An editor for specifying boundary (an array of 4 points) over an selected image.
* **com.vantiq.component.ui.DirectionDistanceEditor** - An editor for editing "distance" or "direction" property of a TrackingRegion.
* **com.vantiq.component.ui.ConvertCoordinates** - An editor for editing "convertionSepcification" property of a TrackingRegion.
* **com.vantiq.component.ui.TrackingRegionEditor** - A Client component to create/view/edit a tracking region. It use above three client components to edit its properties.  
```

File: TrackingRegion/projects/TrackingRegion.json [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+    "source" : "client/TestTrackingRegion",
+    "target" : "document/entranceSample.png"
+  } ],
+  "name" : "TrackingRegion",
+  "options" : {
+    "description" : "This project  contains 4 client components that you can use in your clients.\n\n### Client Components: ###\n* com.vantiq.component.ui.BoundaryEditor - A eitor for specifying boundary (an array of 4 points) over an selected image.\n* com.vantiq.component.ui.DirectionDistanceEditor - A editor for editing \"distance\" or \"direction\" property of a TrackingRegion.\n* com.vantiq.component.ui.ConvertCoordinates - A editor for editing \"convertionSepcification\" property of a TrackingRegion.\n* com.vantiq.component.ui.TrackingRegionEditor - A Client component to create/view/edit a tracking region. It use above three client components to edit its properties.  \n\n### Using the Tracking Region Editor ###\nYou can run com.vantiq.component.ui.TrackingRegionEditor as a normal Client to create/edit/view tracking regions.\nYou can also embed the editor as a component to your Client. See TestTrackingRegion as an example. The TrackingRegionEditor component's toolbar buttons are hidden by setting its hideToolbar property to true.",
```
Comment (rblumer4) [inline TrackingRegion/projects/TrackingRegion.json]: This is the same as the README.md, so need to fix the same typos & add formatting here.

**Review by rblumer4 — CHANGES_REQUESTED** [review]
Did not review the code.
Just reviewed the README and ran the clients.   Sent one stacktrace from ConvertCoordinates via email...

File: TrackingRegion/README.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    * **points** -- a list of two points
+    * **distance** -- the distance between these two points
+* **direction** -- this is an Object containing the following properties
+    * **points** -- a list of two points
+    * **direction** -- the direction (compass degrees between 0 and 360) that movement from the
+first point to the second represents.
+* **converstionOnly** -- a boolean 
+* **conversionSpecification** -- this is an Object defines the conversion between 2 coordinates. It containing the following properties
+    * **imageCoordinates** -- A set of 4 points in input coordinates with are 4 corners of image used by **boundary**, **distance**, and **direction**.
+    * **convertedCoordinates** -- A set of 4 porints converted from 4 points of **imageCoordinates" on the output coordinates.  Output coordinates can be set of latitude and longitude coordinates on a map.
```
Comment (rblumer4) [inline TrackingRegion/README.md]: typo: porints to points
set -> a set
```suggestion
    * **convertedCoordinates** -- A set of 4 points converted from 4 points of **imageCoordinates" on the output coordinates.  Output coordinates can be a set of latitude and longitude coordinates on a map.
```

File: TrackingRegion/README.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+* **name** -- this is the name of the tracking region.
+* **boundary** -- this is an Object that contains a list of (at most 4) points that comprise the boundary of the region.
+* **distance** -- this is an Object containing the following properties:
+    * **points** -- a list of two points
+    * **distance** -- the distance between these two points
+* **direction** -- this is an Object containing the following properties
+    * **points** -- a list of two points
+    * **direction** -- the direction (compass degrees between 0 and 360) that movement from the
+first point to the second represents.
+* **converstionOnly** -- a boolean
```
Comment (rblumer4) [inline TrackingRegion/README.md]: typo?
```suggestion
* **conversionOnly** -- a boolean 
```

File: TrackingRegion/projects/TrackingRegion.json [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
...### Client Components: ###\n* **com.vantiq.component.ui.BoundaryEditor** - An editor for specifying boundary (an array of 4 points) over an selected image.\n\n![Boundary Editor](docs/boundary_editor.JPG)\n\n* **com.vantiq.component.ui.DirectionDistanceEditor** - An editor for editing \"distance\" or \"direction\" property of a TrackingRegion.\n\n![Distance/Direction Editor](docs/distance_editor.JPG)\n* **com.vantiq.component.ui.ConvertCoordinates** - An editor for editing \"convertionSepcification\" property of a TrackingRegion.\n\n![ConvertCoordinates Editor](docs/convert_editor.JPG)\n\n* **com.vantiq.component.ui.TrackingRegionEditor** - A Client component to create/view/edit a tracking region. It use above three client components to edit its properties.  \n\n\n![TrackingRegion Editor](docs/region_editor.JPG)\n\n### Using the Tracking Region Editor ###\nYou can run com.vantiq.component.ui.TrackingRegionEditor as a normal Client to create/edit/view tracking regions.\n\nYou can also embed the editor as a component to your Client. See _TestTrackingRegion_ as an example. The TrackingRegionEditor component's toolbar buttons are hidden by setting its _hideToolbar_ property to `true`.",
```
Comment (rblumer4) [inline TrackingRegion/projects/TrackingRegion.json]: typo: porints -> points
set -> a set
converstionOnly -> conversionOnly
```suggestion
    "description" : "This project  contains 4 client components that provides a grapical interface for editing a Tracking Region.\n\n### About Tracking Region ###\nA tracking region represents a named area in a coordinate space.\nTracking regions have the following properties:\n\n* **name** -- this is the name of the tracking region.\n* **boundary** -- this is an Object that contains a list of (at most 4) points that comprise the boundary of the region.\n* **distance** -- this is an Object containing the following properties:\n    * **points** -- a list of two points\n    * **distance** -- the distance between these two points\n* **direction** -- this is an Object containing the following properties\n    * **points** -- a list of two points\n    * **direction** -- the direction (compass degrees between 0 and 360) that movement from the\nfirst point to the second represents.\n* **conversionOnly** -- a boolean \n* **conversionSpecification** -- this is an Object defines the conversion between 2 coordinates. It containing the following properties\n    * **imageCoordinates** -- A set of 4 points in input coordinates with are 4 corners of image used by **boundary**, **distance**, and **direction**.\n    * **convertedCoordinates** -- A set of 4 points converted from 4 points of **imageCoordinates\" on the output coordinates.  Output coordinates can be a set of latitude and longitude coordinates on a map.\n\n### Client Components: ###\n* **com.vantiq.component.ui.BoundaryEditor** - An editor for specifying boundary (an array of 4 points) over an selected image.\n\n![Boundary Editor](docs/boundary_editor.JPG)\n\n* **com.vantiq.component.ui.DirectionDistanceEditor** - An editor for editing \"distance\" or \"direction\" property of a TrackingRegion.\n\n![Distance/Direction Editor](docs/distance_editor.JPG)\n* **com.vantiq.component.ui.ConvertCoordinates** - An editor for editing \"convertionSepcification\" property of a TrackingRegion.\n\n![ConvertCoordinates Editor](docs/convert_editor.JPG)\n\n* **com.vantiq.component.ui.TrackingRegionEditor** - A Client component to create/view/edit a tracking region. It use above three client components to edit its properties.  \n\n\n![TrackingRegion Editor](docs/region_editor.JPG)\n\n### Using the Tracking Region Editor ###\nYou can run com.vantiq.component.ui.TrackingRegionEditor as a normal Client to create/edit/view tracking regions.\n\nYou can also embed the editor as a component to your Client. See _TestTrackingRegion_ as an example. The TrackingRegionEditor component's toolbar buttons are hidden by setting its _hideToolbar_ property to `true`.",
```

**Review by rblumer4 — DISMISSED** [review]
LGTM, found a couple more typos
Did you fix the stacktrace or should I file a bug about that?

**Comment by zhangevan** [comment]
@rblumer4 thanks for the review. the stacktrace was fixed in 1.38 by Steve a while ago.

File: TrackingRegion/README.md (hunk 5 above)
Comment (zhangevan) [reply TrackingRegion/README.md]: yes, typo.

**Review by rblumer4 — APPROVED** [review]
LGTM
stacktrace is now fixed in standalone ConvertCoordinates client

**Pull request merged**